engine/step/step.go

```diff
@@ -72,6 +73,7 @@ var (
 // A step can be configured to evaluate "conditions" before and after the action is performed:
 // - a "skip" condition will be run before and might determine that the step's action can be skipped entirely
 // - a "check" condition will be run after the action, and can control execution flow by examining
+// - a "loop" condition will cause the step to run again
```

Comment:
We should probably mention the condition on which the step is ran again, meaning, when the AND'ed conditions equals to `true`.